integration_tests/test_eval_forecasting.py

```diff
@@ -169,25 +184,130 @@ def test_compute_forecasting_metric() -> None:
-    assert_almost_equal(expected_min_fde, round(metrics["minFDE"], 2), 2)
-    assert_almost_equal(expected_dac, round(metrics["DAC"], 2), 2)
-    assert_almost_equal(expected_miss_rate, round(metrics["MR"], 2), 2)
-    assert_almost_equal(expected_p_min_ade, round(metrics["p-minADE"], 2), 2)
-    assert_almost_equal(expected_p_min_fde, round(metrics["p-minFDE"], 2), 2)
-    assert_almost_equal(expected_p_miss_rate, round(metrics["p-MR"], 2), 2)
-
-    # Case 2
+    expected_metrics: Dict[str, float] = {
+        "minADE": 1.0,
+        "minFDE": 1.0,
+        "DAC": 1.0,
+        "MR": 0.0,
+        "p-minADE": 1.04,
+        "p-minFDE": 1.04,
+        "p-MR": 0.20,
+    }
+    assert_metrics_almost_equal(expected_metrics, metrics)
+
+    # Case 2: Top-2, 1 forecast, probabilities provided
     forecasted_trajectories = {1: [forecasted_2]}
     forecasted_probabilities = {1: [1.0]}
     ground_truth_trajectories = {1: ground_truth}
+    max_n_guesses = 2
+
+    metrics = compute_forecasting_metrics(
+        forecasted_trajectories,
+        ground_truth_trajectories,
+        city_names,
+        max_n_guesses,
+        horizon,
+        miss_threshold,
+        forecasted_probabilities,
+    )
+
+    expected_metrics: Dict[str, float] = {
+        "minADE": 3.23,
+        "minFDE": 11.31,
+        "DAC": 1.0,
+        "MR": 1.0,
+        "p-minADE": 3.23,
+        "p-minFDE": 11.31,
+        "p-MR": 1.0,
+    }
+    assert_metrics_almost_equal(expected_metrics, metrics)
+
+    # Case 3: Top-1, 2 forecast, probabilities provided
+    forecasted_trajectories = {1: [forecasted_1, forecasted_2]}
+    forecasted_probabilities = {1: [0.2, 0.8]}
+    ground_truth_trajectories = {1: ground_truth}
+    max_n_guesses = 1
+
+    metrics = compute_forecasting_metrics(
+        forecasted_trajectories,
+        ground_truth_trajectories,
+        city_names,
+        max_n_guesses,
+        horizon,
+        miss_threshold,
+        forecasted_probabilities,
+    )
+
+    expected_metrics: Dict[str, float] = {
+        "minADE": 3.23,
+        "minFDE": 11.31,
+        "DAC": 1.0,
+        "MR": 1.0,
+        "p-minADE": 3.23,
+        "p-minFDE": 11.31,
+        "p-MR": 1.0,
+    }
+    assert_metrics_almost_equal(expected_metrics, metrics)
+
+    # Case 4: Top-2, 2 forecast, probabilities provided (not normalized)
+    forecasted_trajectories = {1: [forecasted_1, forecasted_2]}
+    forecasted_probabilities = {1: [0.3, 0.2]}
+    ground_truth_trajectories = {1: ground_truth}
+    max_n_guesses = 2
+
+    metrics = compute_forecasting_metrics(
+        forecasted_trajectories,
+        ground_truth_trajectories,
+        city_names,
+        max_n_guesses,
+        horizon,
+        miss_threshold,
+        forecasted_probabilities,
+    )
+
+    expected_metrics: Dict[str, float] = {
+        "minADE": 1.0,
+        "minFDE": 1.0,
+        "DAC": 1.0,
+        "MR": 0.0,
+        "p-minADE": 1.16,
```

Comment:
these numbers seem a bit hard to work out by hand -- can we come up with a simple example, e.g. straight lines or something, where we can work it out by hand?

Reply:
I guess the final metric values need not be easy to work out by hand as long as the test example is self-explanatory. The test example covers a few cases which straight-line examples will miss.

Reply:
Actually, this particular case can even be worked out by hand. The forecast and ground truth are 1.0m apart in each frame. This is not any more complicated than straight lines.
The other case where the forecast is a turning trajectory might not be that straightforward.